src/service.rs

```diff
@@ -0,0 +1,52 @@
+//! HTTP gateway service for Holochain
+
+use std::net::{IpAddr, SocketAddr};
+
+use axum::{routing::get, Router};
+use tokio::net::TcpListener;
+
+use crate::{routes::healthz, tracing::initialize_tracing_subscriber};
+
+/// Core Holochain HTTP gateway service
+#[derive(Debug)]
+pub struct HcHttpGatewayService {
+    address: SocketAddr,
+    router: Router,
+}
+
+/// Shared application state
+#[derive(Debug, Clone)]
+pub struct AppState {}
+
+impl HcHttpGatewayService {
+    /// Create a new service instance bound to the given address and port
+    pub fn new(address: impl Into<IpAddr>, port: u16) -> Self {
+        let address = SocketAddr::new(address.into(), port);
+
+        let router = Router::new()
+            .route("/healthz", get(healthz))
+            .with_state(AppState {});
+
+        HcHttpGatewayService { router, address }
+    }
+
+    /// Get the socket address the service is configured to use
+    pub fn address(&self) -> SocketAddr {
+        self.address
+    }
+
+    /// Start the HTTP server and run until terminated
+    pub async fn run(self) -> anyhow::Result<()> {
```

Comment:
Avoid using `anyhow` in library code. It's convenient but really hard to use with other error handling libraries in upstream code. You've set up `thiserror` already. Recommend defining a result type that uses the error for it and use that here instead